src/main/java/org/opensearch/geospatial/stats/upload/UploadStats.java

```diff
@@ -47,6 +38,19 @@ public static UploadStats getInstance() {
         totalAPICount = new CounterMetric();
     }
 
+    /**
+     * Get UploadStats from {@link StreamInput}.
+     * @param input contains {@link UploadStats} in serialized form
+     * @return UploadStats instance
+     * @throws IOException if cannot read {@link UploadStats} from given input
+     */
+    public static UploadStats fromStreamInput(StreamInput input) throws IOException {
```

Comment:
Im wondering if this makes more sense as a constructor of UploadStats. Did you think about this?

Reply:
Right. I thought about it. I chose this over other because of "throws IOException". If it is going to just save the parameter as member input, i would have went with constructor format. What do you think?

Reply:
Hmm not sure I understand. Is it bad practice to have an exception thrown from constructor?